modules/core/02-client/migrations/v7/solomachine.go

```diff
@@ -0,0 +1,263 @@
+package v7
+
+import (
+	ics23 "github.com/confio/ics23/go"
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/cosmos/ibc-go/v6/modules/core/exported"
+)
+
+// NOTE: this is a mock implmentation for exported.ClientState. This implementation
+// should only be registered on the InterfaceRegistry during cli command genesis migration.
+// This implementation is only used to successfully unmarshal the previous solo machine
+// client state and consensus state and migrate them to the new implementations. When the proto
+// codec unmarshals, it calls UnpackInterfaces() to create a cached value of the any. The
+// UnpackInterfaces function for IdenitifiedClientState will attempt to unpack the any to
+// exported.ClientState. If the solomachine v1 type is not registered against the exported.ClientState
+// the unmarshal will fail. This implementation will panic on every interface function.
+// The same is done for the ConsensusState.
+
+// Interface implementation checks.
+var (
+	_, _ codectypes.UnpackInterfacesMessage = &ClientState{}, &ConsensusState{}
+	_    exported.ClientState               = (*ClientState)(nil)
+	_    exported.ConsensusState            = &ConsensusState{}
+)
+
+func RegisterInterfaces(registry codectypes.InterfaceRegistry) {
```

Comment:
nit, may as well add a godoc

```suggestion

// RegisterInterfaces registers the solomachine v2 ClientState and ConsensusState types in the interface registry.
func RegisterInterfaces(registry codectypes.InterfaceRegistry) {
```